File: java/src/main/java/com/cartenz/core/api/BaseApiDao.java
package com.cartenz.core.api;


import java.util.List;

public class BaseApiDao<T> {
    public T data;
    public int code;
    public String message;
    public List<Errors> errors;

    public static class Errors {
        public int code;
        public int itemCode;
    }
}

File: cartenz/src/main/java/com/cartenz/cartenz/api/BaseApiDao.kt
package com.cartenz.kotlin_core.api

open class BaseApiDao<T> {
    var data: T? = null
    var code: Int = 0
    var message: String? = null
    var errors: List<Errors>? = null

    class Errors {
        var code: Int = 0
        var itemCode: Int = 0
    }
}